data-pipeline/src/trace_exporter/mod.rs

```diff
@@ -425,6 +425,33 @@ impl TraceExporter {
         }
     }
 
+    /// !!! This function is only for testing purposes !!!
+    /// This function waits the agent info to be ready by checking the agent_info state.
+    /// It will only return Ok after the agent info has been fetched at least once or Err if timeout
+    /// has been reached
+    ///
+    /// In production:
+    /// 1) We should not synchronously wait for this to be ready before sending traces
+    /// 2) It's not guaranteed to not block forever, since the /info endpoint might not be
+    ///    available.
+    ///
+    /// The `send`` function will check agent_info when running, which will only be available if the
+    /// fetcher had time to reach to the agent.
+    /// Since agent_info can enable CSS computation, waiting for this during testing can make
+    /// snapshots non-determinitic.
```

Comment:
It was removed because the function it gated was no longer necessary, but I'd suggest adding back the [test-utils feature to data-pipeline](https://github.com/DataDog/libdatadog/pull/897/commits/a559948708c34f50fd3051526df57fac10b16a33) and gating this function behind it.

Reply:
i added a feature gate